server/mscalendar/mscalendar.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package mscalendar
+
+import (
+	"github.com/mattermost/mattermost-server/v5/model"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/config"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/remote"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/store"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils/bot"
+)
+
+type MSCalendar interface {
+	Availability
+	Calendar
+	EventResponder
+	Subscriptions
+	Users
+}
+
+// Dependencies contains all API dependencies
+type Dependencies struct {
+	Logger    bot.Logger
+	PluginAPI PluginAPI
+	Poster    bot.Poster
+	Remote    remote.Remote
+	Store     store.Store
+}
+
+type PluginAPI interface {
+	GetMattermostUser(mattermostUserID string) (*model.User, error)
+	GetMattermostUserByUsername(mattermostUsername string) (*model.User, error)
+	GetMattermostUserStatus(userID string) (*model.Status, error)
+	GetMattermostUserStatusesByIds(userIDs []string) ([]*model.Status, error)
+	IsSysAdmin(mattermostUserID string) (bool, error)
+	UpdateMattermostUserStatus(userID, status string) (*model.Status, error)
+}
+
+type Env struct {
+	*config.Config
+	*Dependencies
+}
+
+type mscalendar struct {
+	Env
+
+	actingUser *User
+	client     remote.Client
+}
+
+func New(env Env, actingMattermostUserID string) MSCalendar {
+	if actingMattermostUserID == "" {
+		actingMattermostUserID = env.Config.BotUserID
```

Comment:
Yes I do. First there's a philosophical reason - if an action is taken it's taken by someone.

Secondly, there's a practical reason. It might be clear to check the value, and error if it's invalid (e.g. ""). Then the error needs to be handled, etc. I am not 100% convinced either way, but I also don't see it as too big a deal. LMK if you do,

The `withActingUser[Expanded]` is used everywhere and not having a UserID will cause it to fail.

So...